src/libraries/Common/src/Extensions/ChangeCallbackRegistrar.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.Extensions.Internal
+{
+    internal static partial class ChangeCallbackRegistrar
+    {
+        /// <summary>
+        /// Registers for a callback that will be invoked when the entry has changed. <see cref="Primitives.IChangeToken.HasChanged"/>
+        /// MUST be set before the callback is invoked.
+        /// </summary>
+        /// <param name="callback">The callback to invoke.</param>
+        /// <param name="state">State to be passed into the callback.</param>
+        /// <param name="Token">The <see cref="CancellationToken"/> to invoke the callback with.</param>
+        /// <param name="onFailure">The action to execute when an <see cref="ObjectDisposedException"/> is thrown. Should be used to set the IChangeToken's ActiveChangeCallbacks property to false.</param>
+        /// <param name="onFailureState">The state to be passed into the <paramref name="onFailure"/> action.</param>
+        /// <returns>The <see cref="CancellationToken"/> registration.</returns>
+        internal static IDisposable UnsafeRegisterChangeCallback<T>(Action<object?> callback, object? state, CancellationToken Token, Action<T> onFailure, T onFailureState)
+        {
+#if NETCOREAPP || NETSTANDARD2_1
+            try
+            {
+                return Token.UnsafeRegister(callback, state);
```

Comment:
```suggestion
        internal static IDisposable UnsafeRegisterChangeCallback<T>(Action<object?> callback, object? state, CancellationToken token, Action<T> onFailure, T onFailureState)
        {
#if NETCOREAPP || NETSTANDARD2_1
            try
            {
                return token.UnsafeRegister(callback, state);
```

Parameters and local variables are camelCased.